cli/translation-cli/src/main/kotlin/net/thunderbird/cli/translation/TranslationCli.kt

```diff
@@ -0,0 +1,73 @@
+package net.thunderbird.cli.translation
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.double
+import net.thunderbird.cli.translation.net.Language
+import net.thunderbird.cli.translation.net.WeblateClient
+
+const val TRANSLATED_THRESHOLD = 70.0
+
+class TranslationCli(
+    private val client: WeblateClient = WeblateClient(),
+    private val configurationsFormatter: ResourceConfigurationsFormatter = ResourceConfigurationsFormatter(),
+    private val supportedLanguagesFormatter: SupportedLanguagesFormatter = SupportedLanguagesFormatter(),
+) : CliktCommand(
+    name = "translation",
+    help = "Translation check CLI",
+) {
+    private val token: String by option(
+        help = "Weblate API token",
+    ).required()
+
+    private val threshold: Double by option(
+        help = "Threshold for translation completion",
+    ).double().default(TRANSLATED_THRESHOLD)
+
+    private val printAll: Boolean by option(
+        help = "Print code example",
+    ).flag()
+
+    override fun run() {
+        val languages = client.loadLanguages(token)
+        val filteredLanguages = filterByThreshold(languages, threshold)
+        val androidVersionCodes = mapToAndroidVersionCodes(filteredLanguages)
+
+        echo("\nLanguages that are translated above the threshold of ($threshold%): ${androidVersionCodes.size}")
+        echo()
+        echo(androidVersionCodes.joinToString(","))
+        if (printAll) {
+            echo("--------------------------------------------------------------")
+            echo(configurationsFormatter.format(androidVersionCodes))
+            echo("--------------------------------------------------------------")
+            echo(supportedLanguagesFormatter.format(androidVersionCodes))
+            echo("--------------------------------------------------------------")
+        }
+        echo()
+    }
+
+    private fun filterByThreshold(languages: List<Language>, threshold: Double): List<Language> {
+        return languages.filter { it.translatedPercent >= threshold }
+    }
+
+    private fun mapToAndroidVersionCodes(languages: List<Language>): List<String> {
+        return languages.map { mapToAndroidVersionCode(it.code) }
+            .sorted()
+    }
+
+    private fun mapToAndroidVersionCode(code: String): String {
```

Comment:
In addition, Weblate seems to map "nb_NO" to "nb", and "ta" to "ta_IN". At least those are the ones that affect us right now.

Reply:
Android does allow both, so `nb` and `nb_rNO` are valid. As we prepoulated weblate with already existing translations, they are taken as is. I guess some of that is inherited from transifex.

I don't know how it will be for new languages, but from the source code I can't see other mappings.

Reply:
I'll need to look into the weblate api, if there is an easy way to receive all the mappings from there